lib/dao.js

```diff
@@ -1183,8 +1191,9 @@ DataAccessObject._coerce = function (where) {
       }
     } else {
       if (val != null) {
-        if (!((operator === 'like' || operator === 'nlike') &&
-          val instanceof RegExp)) {
+        if (val instanceof RegExp) {
+          // do not coerce regex literals/objects
+        } else if (!((operator === 'like' || operator === 'nlike') && val instanceof RegExp)) {
```

Comment:
Does `like` and `nlike` use regex strings? Are we trying to say:

- if operator !== like
- or operator !== nlike
- and val is a regex object
- then coerce the value based on its data type in the model definition?